### PR TITLE
Add support for `--fullscreen`, `--maximized`

### LINKED ISSUE #5801: Fullscreen toggle commandline parameter
# Description of the new feature/enhancement

A command-line parameter that can toggle fullscreen status; initially since you can't modify existing WT sessions (to add or remove tabs) yet you wouldn't be able to use it to turn FS off, but you could use it when running a WT script to open several tabs to make those fullscreen.

# Proposed technical implementation details (optional)

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Adds two new flags to the `wt.exe` alias:

* `--maximized,-M`: Launch the new Terminal window maximized. This flag cannot be combined with `--fullscreen`.
* `--fullscreen,-F`: Launch the new Terminal window fullscreen. This flag cannot be combined with `--maximized`.

## References
* This builds on the work done in #6060.
* The cmdline args megathread: #4632

## PR Checklist
* [x] Closes #5801
* [x] I work here
* [ ] Tests added/passed
* [n/a] Requires documentation to be updated

## Detailed Description of the Pull Request / Additional comments

* I had to move the commandline arg parsing up a layer from `TerminalPage` to `AppLogic`, because `AppLogic` controls the Terminal's settings, including launch mode settings. This seems like a reasonable change, to put both the settings from the file and the commandline in the same place.
  - **Most of the diff is that movement of code**

* _"What happens when you try to pass both flags, like `wtd -M -F new-tab`?"_:
image

## Validation Steps Performed
* Ran a bunch of commandlines to see what happened.

### CODE REVIEW

**Comment by zadjii-msft** [comment]
Oh, so one the target branch was closed, this can't get re-targeted anymore? That's... good to know

**Comment by DHowett** [comment]
I just saw that GitHub had a new feature that automatically changed branch targeting if you did this, so I wonder why it didn’t work

File: src/cascadia/TerminalApp/AppCommandlineArgs.cpp [hunk 1]
```diff
@@ -173,6 +174,18 @@ void AppCommandlineArgs::_buildParser()
...
+    // Maximized and Fullscreen flags
+    auto maximizedCallback = [this](int64_t /*count*/) {
+        _launchMode = winrt::TerminalApp::LaunchMode::MaximizedMode;
+    };
+    auto fullscreenCallback = [this](int64_t /*count*/) {
+        _launchMode = winrt::TerminalApp::LaunchMode::FullscreenMode;
+    };
+    auto maximized = _app.add_flag_function("-M,--maximized", maximizedCallback, RS_A(L"CmdMaximizedDesc"));
+    auto fullscreen = _app.add_flag_function("-F,--fullscreen", fullscreenCallback, RS_A(L"CmdFullscreenDesc"));
+    maximized->excludes(fullscreen);
```
Comment (miniksa) [inline src/cascadia/TerminalApp/AppCommandlineArgs.cpp:186]: You only need this to exclude the one way, right? What happens if you `fullscreen->excludes(maximized)` also? Nothing? Explosions?

File: src/cascadia/TerminalApp/AppCommandlineArgs.cpp [hunk 2]
```diff
@@ -157,6 +157,7 @@ int AppCommandlineArgs::_handleExit(const CLI::App& command, const CLI::Error& e
 // - <none>
 void AppCommandlineArgs::_buildParser()
 {
+    // -v,--version
```
Comment (miniksa) [inline src/cascadia/TerminalApp/AppCommandlineArgs.cpp:160]: You commented this one with the arguments and you commented the below one with a textual explanation....

Consistency? Maybe do both for both? 

```
//-v,--version: Displays version info
//-M,--maximized: Maximizes the window on launch
//-F,--fullscreen: Fullscreens the window on launch
```

File: src/cascadia/TerminalApp/AppCommandlineArgs.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/AppCommandlineArgs.cpp:186]: Yep that's right, it only needs to be done one-way

**Pull request merged**